src/miniscript/types/extra_props.rs

```diff
@@ -1021,6 +1003,46 @@ impl Property for ExtData {
     }
 }
 
+// Function to pass to sort_by. Sort by (satisfaction cost - dissatisfaction cost).
+//
+// We sort by (satisfaction cost - dissatisfaction cost) to make a worst-case (the most
+// costy satisfactions are satisfied, the most costy dissatisfactions are dissatisfied).
+//
+// Args are of form: (<count_sat>, <count_dissat>)
+fn sat_minus_dissat<'r, 's>(
```

Comment:
In cb4c0376154c3699155d2c546f5a1356b30ed8e6:

I think all these functions should be defined at the top of the function that they're used in.

Reply:
I would like these functions' visibility to match the only scope where they're used, since they are essentially helpers for the code in that scope.

Reply:
This isn't about being C-like -- you can't even do what I'm suggesting in C.

Reply:
I have no strong preferences here. 
If the name is descriptive and is being used multiple times in different methods(I think it is the case for some of the functions here). It makes sense to have it at the bottom.  

On the other hand, if it is just a helper function, it makes sense to have it a function near the body or even a sub-function inside the main function. The readers can cleanly separate these things.